tensorflow/contrib/tensorrt/test/tf_trt_integration_test_base.py

```diff
@@ -514,9 +520,20 @@ def _Test(self):
 
   use_optimizer_options = [False, True]
   dynamic_engine_options = [False, True]
-  for (use_optimizer, precision_mode, dynamic_engine) in itertools.product(
-      use_optimizer_options, PRECISION_MODES, dynamic_engine_options):
-    if IsQuantizationMode(precision_mode):
+  use_calibration_options = [False, True]
+  opts = itertools.product(use_optimizer_options, PRECISION_MODES,
+                           dynamic_engine_options, use_calibration_options)
+  for (use_optimizer, precision_mode, dynamic_engine, use_calibration) in opts:
+    conversion = "OptimizerConversion" if use_optimizer else "ToolConversion"
+    engine_type = ("DynamicEngine" if dynamic_engine else "StaticEngine")
+    test_name = "%s_%s_%s" % (conversion, precision_mode, engine_type)
+    run_params = RunParams(
+        use_optimizer=use_optimizer,
+        precision_mode=precision_mode,
+        dynamic_engine=dynamic_engine,
+        test_name=test_name,
+        use_calibration=use_calibration)
+    if NeedsCalibration(run_params):
```

Comment:
Maybe add:
```
if precision_mode != 'INT8' and use_calibration:
  continue
```
So we don't run tests for FP32/FP16 twice.

Reply:
Also based on what you said, if `precision_mode == 'INT8' and not use_calibration`, engine building should fail, meaning that all existing tests will fail, is that right?

If so we may want to let `ShouldRunTest()` return `False` if `precision_mode == 'INT8' and not use_calibration` by default, and override it in your newly added tests (which you already did).

Reply:
Oh right, that makes sense. I will also revert `NeedsCalibration()` back to `IsQuantizationMode()` because both methods of using INT8 require special considerations.

Reply:
> Also based on what you said, if `precision_mode == 'INT8' and not use_calibration`, engine building should fail, meaning that all existing tests will fail, is that right?
> If so we may want to let `ShouldRunTest()` return `False` if `precision_mode == 'INT8' and not use_calibration` by default, and override it in your newly added tests (which you already did).

I will also have to modify every test that overrides `ShouldRunTest()` to exclude INT8 w/o calibration, correct?